modules/programs/ssh.nix

```diff
@@ -0,0 +1,155 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.ssh;
+
+  yn = flag: if flag then "yes" else "no";
+
+  hostModule = types.submodule ({...}: {
+    options = {
+
+      port = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = "Specifies port number to connect on remote host.";
+      };
+
+      forwardX11 = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Specifies whether X11 connections will be automatically redirected
+          over the secure channel and DISPLAY set.
+        '';
+      };
+
+      forwardX11Trusted = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Specifies whether remote X11 clients will have full access to the
+          original X11 display.
+        '';
+      };
+
+      identitiesOnly = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Specifies that ssh should only use the authentication identity
+          explicitly configured in the .ssh/config files or passed on the
+          ssh command-line, even if ssh-agent offers more identities.
+        '';
+      };
+
+      identityFile = mkOption {
+        type = types.nullOr types.string;
+        default = null;
+        description = ''
+          Specifies a file from which user's identity is read.
+        '';
+      };
+
+      user = mkOption {
+        type = types.nullOr types.string;
+        default = null;
+        description = "Specifies the user to log in as.";
+      };
+
+      hostname = mkOption {
+        type = types.nullOr types.string;
+        default = null;
+        description = "Specifies the real host name to log into.";
+      };
+
+      serverAliveInterval = mkOption {
+        type = types.int;
+        default = 0;
+        description =
+          "Set timeout in seconds after which response will be requested.";
+      };
+
+      checkHostIP = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Check that host IP address in the known_hosts file.";
+      };
+    };
+  });
+
+  hostStr = host: cf: concatStringsSep "\n" (
+    ["host ${host}"]
+    ++ optionals (cf.port != null)         ["  port ${toString cf.port}"]
```

Comment:
Instead of, e.g.,
```
optionals (cf.port != null)         ["  port ${toString cf.port}"]
```
could use
```
optional (cf.port != null)         "  port ${toString cf.port}"
```
Not a huge difference, exactly, but maybe a little simpler :-)